spec/inputs/syslog_spec.rb

```diff
@@ -66,13 +72,103 @@
 
       events = event_count.times.collect { queue.pop }
 
+      # important to shutdown here before any assertion so that the pipeline + socket
+      # cleanups are correctly done before any potential spec error that would result
+      # in aborting execution and not doing the cleanup.
+      pipeline.shutdown
+      t.join
+
       insist { events.length } == event_count
       event_count.times do |i|
-        insist { events[i]["tags"] } == ["_grokparsefailure_sysloginputplugin"]
+        insist { events[i]["tags"] } == ["_grokparsefailure_sysloginput"]
+      end
+    end
+  end
+
+  describe "should properly handle locale and timezone" do
+    port = 5511
+    event_count = 10
+
+    config <<-CONFIG
+      input {
+        syslog {
+          type => "blah"
+          port => #{port}
+          locale => "en"
+          timezone => "UTC"
+        }
+      }
+    CONFIG
+
+    input do |pipeline, queue|
+      t = Thread.new { pipeline.run }
+      sleep 0.1 while !pipeline.ready?
+
+      socket = Stud.try(5.times) { TCPSocket.new("127.0.0.1", port) }
+      event_count.times do |i|
+        socket.puts("<164>Oct 26 15:19:25 1.2.3.4 %ASA-4-106023: Deny udp src DRAC:10.1.2.3/43434 dst outside:192.168.0.1/53 by access-group \"acl_drac\" [0x0, 0x0]")
+      end
+      socket.close
+
+      events = event_count.times.collect { queue.pop }
+
+      # important to shutdown here before any assertion so that the pipeline + socket
+      # cleanups are correctly done before any potential spec error that would result
+      # in aborting execution and not doing the cleanup.
+      pipeline.shutdown
+      t.join
+
+      insist { events.length } == event_count
+      events.each do |event|
+        insist { event["@timestamp"].to_iso8601 } == "#{Time.now.year}-10-26T15:19:25.000Z"
       end
     end
   end
 
+  describe "should properly handle no locale and no timezone" do
+    port = 5511
+
+    config <<-CONFIG
+      input {
+        syslog {
+          type => "blah"
+          port => #{port}
+        }
+      }
+    CONFIG
+
+    input do |pipeline, queue|
+      t = Thread.new { pipeline.run }
+      sleep 0.1 while !pipeline.ready?
+
+      socket = Stud.try(5.times) { TCPSocket.new("127.0.0.1", port) }
+      socket.puts("<164>Oct 26 15:19:25 1.2.3.4 %ASA-4-106023: Deny udp src DRAC:10.1.2.3/43434 dst outside:192.168.0.1/53 by access-group \"acl_drac\" [0x0, 0x0]")
+      socket.close
```

Comment:
I saw this used twice, we could move this into a `let`
```
"<164>Oct 26 15:19:25 1.2.3.4 %ASA-4-106023: Deny udp src DRAC:10.1.2.3/43434 dst outside:192.168.0.1/53 by access-group \"acl_drac\" [0x0, 0x0]"
```
other reference: https://github.com/logstash-plugins/logstash-input-syslog/pull/1/files#diff-6bffb5d7bae5d75fb475af76a37a54eaR109